src/consensus/block_body_validation.py

```diff
@@ -155,231 +152,247 @@ async def validate_block_body(
     if height <= constants.INITIAL_FREEZE_PERIOD and block.transactions_generator is not None:
         return Err.INITIAL_TRANSACTION_FREEZE, None
 
-    if block.transactions_generator is not None:
-        # Get List of names removed, puzzles hashes for removed coins and conditions crated
-        if cached_cost_result is not None:
-            result: Optional[CostResult] = cached_cost_result
+    if constants.NETWORK == NetworkType.MAINNET:
+        if block.transactions_generator is not None:
+            if len(bytes(block.transactions_generator)) > constants.MAX_GENERATOR_SIZE:
+                return Err.PRE_SOFT_FORK_MAX_GENERATOR_SIZE, None
+            else:
+                return None, None
```

Comment:
We should return an error here instead of None

Reply:
in case of else: ? - there's no error